client/ladder/action-creators.ts

```diff
@@ -17,37 +18,46 @@ export function getRankings(
   spec: RequestHandlingSpec<void>,
 ): ThunkAction {
   return abortableThunk(spec, async dispatch => {
-    const fetchTime = new Date()
-    const lastFetchTime = lastFetchTimeByMatchmakingType.get(matchmakingType) ?? new Date(0)
+    // `performance.now()` can return value that's less than our cache time, so we floor it to that.
```

Comment:
This isn't really a floor, if you want that then you should use `Math.max`. I don't really understand this code though, the constant is used to check for the difference between the current time and the last time. This feels like a hacky workaround to avoid dealing with the fact that the last time can be undefined, and the code would be easier to understand if you just dealt with that directly instead.

Reply:
I used "floor" here in the sense that this is the lowest value that the fetch time can be now.

The reason why I did this is because `performance.now()` returns the time since the beginning of the current document's lifetime (unlike `Date.now()` which returns time since 1971 or whatever). This means that `performance.now()` can return, for example, 20 seconds, while our cache time is 1 minute, which throws a wrench in our condition below where we assume that all requests made during initial cache time period are already cached.

I could rewrite the condition to treat any request made during initial cache time (e.g. `fetchTime < LADDER_RANKINGS_CACHE_TIME_MS`) as a way to always make the request, but then we're losing benefits of caching during that period. It's also possible that I'm overly complicating this and am missing something obvious :d

Reply:
I get why you're doing it, it's not a correct thing to do. The correct thing to do is to properly handle the case that it has not been fetched, and not use 0 as a magic value to indicate this.